### PR TITLE
Added new service and tests for it

### PR DESCRIPTION
**Pull request by MarianLabensky**

### CODE REVIEW

File: EPlast/EPlast.BLL/Services/ActiveMembership/UserDatesService.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            if(userDto != null)
+            {
+                UserMembershipDates userMembershipDates = await _repoWrapper.UserMembershipDates.GetFirstOrDefaultAsync(umd => umd.UserId == userId);
+                
+                if (userMembershipDates != null)
+                {
+                    return _mapper.Map<UserMembershipDatesDTO>(userMembershipDates);
+                }
+            }
+            return null;
```
Comment (voroniak) [inline EPlast/EPlast.BLL/Services/ActiveMembership/UserDatesService.cs:61]: return null is considered a bad practice

Comment (voroniak) [inline EPlast/EPlast.BLL/Services/ActiveMembership/UserDatesService.cs:61]: return null is considered a bad practice

Comment (MarianLabensky) [reply EPlast/EPlast.BLL/Services/ActiveMembership/UserDatesService.cs]: Ok, I changed it, now it throw exception instead return null

Comment (voroniak) [reply EPlast/EPlast.BLL/Services/ActiveMembership/UserDatesService.cs]: It's better to fix  5 Code Smells too.

Comment (MarianLabensky) [reply EPlast/EPlast.BLL/Services/ActiveMembership/UserDatesService.cs]: Done))

File: EPlast/EPlast.Tests/Services/ActiveMembership/UserDatesServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+            Assert.ThrowsAsync<InvalidOperationException>(() => userDatesService.GetUserMembershipDatesAsync(" "));
+
+            _userManagerService.Verify(f => f.FindByIdAsync(It.IsAny<string>()));
+            _repoWrapper.Verify(f => f.UserMembershipDates.GetFirstOrDefaultAsync(It.IsAny<Expression<Func<UserMembershipDates, bool>>>(),
+            It.IsAny<Func<IQueryable<UserMembershipDates>, IIncludableQueryable<UserMembershipDates, object>>>()));
+        }
+        /// <summary>
+        /// ///////////////
+        /// </summary>
+        /// <returns></returns>
```
Comment (voroniak) [inline EPlast/EPlast.Tests/Services/ActiveMembership/UserDatesServiceTests.cs:153]: ???

**Review by voroniak — CHANGES_REQUESTED** [review]

File: EPlast/EPlast.Tests/Services/ActiveMembership/UserDatesServiceTests.cs (hunk 2 above)
Comment (MarianLabensky) [reply EPlast/EPlast.Tests/Services/ActiveMembership/UserDatesServiceTests.cs]: I just forgot about it, now it`s fixed

**Review by voroniak — APPROVED** [review]

**Review by TonyWebber — APPROVED** [review]

**Review by MykhailoKlapchuk — APPROVED** [review]

**Pull request merged**